### PR TITLE
Fix bug with construct_curr_info

### PR DESCRIPTION
**Pull request by ervteng**
- `construct_curr_info` doesn't work with visual observations as the initialization of the BrainInfo's list is wrong. This fixes it.

### CODE REVIEW

File: ml-agents/mlagents/trainers/tests/test_rl_trainer.py [hunk 1]
```diff
@@ -47,7 +47,8 @@ def create_mock_policy():
 
 @mock.patch("mlagents.trainers.rl_trainer.RLTrainer.add_policy_outputs")
 @mock.patch("mlagents.trainers.rl_trainer.RLTrainer.add_rewards_outputs")
-def test_rl_trainer(add_policy_outputs, add_rewards_outputs):
+@pytest.mark.parametrize("num_vis_obs", [0, 1], ids=["", "viz"])
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/tests/test_rl_trainer.py]: is it possible to have the num_vis_obs also be 2 ? Or is that too much work ?

Comment (chriselion) [inline ml-agents/mlagents/trainers/tests/test_rl_trainer.py]: What does `ids=...` do here? Doesn't seem to be used.

Comment (chriselion) [reply ml-agents/mlagents/trainers/tests/test_rl_trainer.py]: nm, I just RTFM'ed, looks like it just affects the test name. I've never needed to override the default behavior.

Comment (ervteng) [reply ml-agents/mlagents/trainers/tests/test_rl_trainer.py]: Added, it's a one-number change

Comment (ervteng) [reply ml-agents/mlagents/trainers/tests/test_rl_trainer.py]: I guess I changed it since it wouldn't be obvious what `test_rl_trainer[X]` meant

**Pull request merged**